catalog/core/catalog-core-solrcommands/src/main/java/org/codice/ddf/commands/solr/ReindexCommand.java

```diff
@@ -0,0 +1,492 @@
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrQuery.ORDER;
+import org.apache.solr.client.solrj.SolrQuery.SortClause;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.params.CursorMarkParams;
+import org.codice.ddf.security.common.Security;
+import org.codice.solr.client.solrj.SolrClient;
+import org.codice.solr.factory.SolrClientFactory;
+
+@Service
+@Command(
+  scope = SolrCommands.NAMESPACE,
+  name = "reindex",
+  description =
+      "Reindexes data from a source collection to current Catalog Framework collection(s)."
+)
+public class ReindexCommand extends SolrCommands {
+  private static final int PAGE_SIZE = 200;
+
+  private static final int WRITE_TXN_SIZE = 100;
+
+  @VisibleForTesting protected static final String EARLY_TIME = "1900-01-01T00:00:00.000Z";
+
+  private Security security = Security.getInstance();
+
+  private String cursorMark = CursorMarkParams.CURSOR_MARK_START;
+
+  private Reader readerThread = null;
+
+  private SolrMetacardClientImpl metacardClient =
+      new SolrMetacardClientImpl(null, null, null, new DynamicSchemaResolver());
+
+  private long totalCount = 0;
+
+  private AtomicLong count = new AtomicLong();
+
+  private long startTime;
+
+  private ThreadPoolExecutor publishExecutor;
+
+  private SolrClient solrjClient = null;
+
+  @Reference private CatalogFramework catalogFramework;
+
+  @Reference private SolrClientFactory clientFactory;
+
+  @Option(
+    name = "-s",
+    aliases = {"--source"},
+    description =
+        "The source Solr system to retrieve data. Should be in the form of http://host:port.",
+    required = true
+  )
+  private String sourceSolrHost;
+
+  @Option(
+    name = "-c",
+    aliases = {"--sourceCollection"},
+    description = "The source collection (or alias) to migrate data from.",
+    required = true
+  )
+  private String sourceCollection;
+
+  @Option(
+    name = "-f",
+    aliases = {"--field"},
+    description =
+        "Field used for date comparisons. Default (Date of indexing): metacard.created_tdt",
+    required = false
+  )
+  private String field = "metacard.created_tdt";
+
+  @Option(
+    name = "-a",
+    aliases = {"--after"},
+    description =
+        "After date used to restrict data to be migrated. Should be in the format: 2019-01-01T00:00:00Z.",
+    required = false
+  )
+  private String afterDate;
+
+  @Option(
+    name = "-b",
+    aliases = {"--before"},
+    description =
+        "Before date used to restrict data to be migrated. Should be in the format: 2019-01-01T00:00:00Z.",
+    required = false
+  )
+  private String beforeDate;
+
+  @Option(
+    name = "-p",
+    aliases = {"--practice"},
```

Comment:
:pencil2: Usually I see this called dry run with a lot of commands.  Maybe `-d` and `--dryrun`.